pkg/specgen/generate/container_create.go

```diff
@@ -642,6 +642,16 @@ func createContainerOptions(rt *libpod.Runtime, s *specgen.SpecGenerator, pod *l
 		options = append(options, libpod.WithHealthCheckOnFailureAction(s.ContainerHealthCheckConfig.HealthCheckOnFailureAction))
 	}
 
+	if healthCheckSet {
+		options = append(options, libpod.WithHealthCheckLogDestination(s.ContainerHealthCheckConfig.HealthLogDestination))
+		options = append(options, libpod.WithHealthCheckMaxLogCount(s.ContainerHealthCheckConfig.HealthMaxLogCount))
+		options = append(options, libpod.WithHealthCheckMaxLogSize(s.ContainerHealthCheckConfig.HealthMaxLogSize))
+	} else {
+		options = append(options, libpod.WithHealthCheckLogDestination(define.DefaultHealthCheckLocalDestination))
+		options = append(options, libpod.WithHealthCheckMaxLogCount(define.DefaultHealthMaxLogCount))
+		options = append(options, libpod.WithHealthCheckMaxLogSize(define.DefaultHealthMaxLogSize))
+	}
+
```

Comment:
This is a bit confusing, I think it is better when defaults are set in libpod on not by specgen.